### PR TITLE
docs: [ANCHOR-126] Simplify the component schemas

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

The transaction event of the `Event Schema.yml` is SEP-31 centric. When we add SEP-24 and SEP-6, the schema can become unstable and requires changes as the project develops. These fields also duplicate the fields defined in the `Platform API.yml` which increases the difficulties for the community to adopt the event schema.

Similar component schemas are merged and unused schemas were removed.

### Why

For simplification and improvement.

### Known limitations

N/A

### CODE REVIEW

**Review by JakeUrban — CHANGES_REQUESTED** [review]
I tried to design the `TransactionEvent` schema to be SEP-generic, can you describe what you think isn't generalized?

The only fields I see that could be confusing are `customers` & `creator`.

We needed `customers` for SEP-31, but in SEP-24 & 6 the customers (`customers.sender` and `customers.receiver`) were intended to be identical.

We needed `creator` for SEP-31 to save the send anchor's public key, but `creator` can also be used for the Stellar wallet. If wallet attribution is used we can put the wallet's home domain in the `creator` field.

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (marcelosalloum) [inline docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:144]: Is this really a good idea to remove this?

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (marcelosalloum) [inline docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:158]: All these timestamps seem important.

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (marcelosalloum) [inline docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:167]: It'd be good to return some Ledger identification, like the transaction or operation id, so the Anchor can reach out to the Ledger for more information.

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (marcelosalloum) [inline docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:190]: Are you sure you want to remove any references from the customers related to this event? This information cannot be retrieved any other way in the future by the Anchor, can it?

**Review by marcelosalloum — COMMENTED** [review]
So you want to force the Anchor to always call `GET /transaction`, right? I think that kind of defeats some of the purpose of the Events approach.

Why don't you unify the payloads of the Event Schema and PlatformAPI? You can update the event so it returns 3 fields: event type, event status, and payload, where the payload is the exact same response you'd get with GET {platform API}/transaction.

This way you can simplify the events, keep parity with the PlatformAPI and keep providing this information to the Anchor.

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (lijamie98) [reply docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:190]: We will add the `transaction` as the payload of the transaction event as you have suggested. 

In that case, all these fields will be available in the transaction event schema.

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (marcelosalloum) [inline docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:14]: To solve the syntax error, I think you need this:
```suggestion
paths: {}
```

**Review by marcelosalloum — COMMENTED** [review]
About these fields you're removing, are all of them available in the Platform API?

I took a look at the platform API and didn't find a few of them, like:
- status_change
- source_account
- destination_account

I think it's ok to drop the status_change, but the others should probably be added to the Transaction Schema. WDYT?

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (lijamie98) [reply docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:14]: I do not think `status_change` is being used and not sure if it has a meaningful use case. The `status_change` should be derived from the log where we record every state of all fields.

The `source_account` and `destination_account` are available in [StellarTransactions.](https://github.com/stellar/java-stellar-anchor-sdk/blob/develop/docs/03%20-%20Implementing%20the%20Anchor%20Server/Communication/Platform%20API.yml#L355)

Comment (marcelosalloum) [reply docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:14]: SGTM!

Do you think it's worth writing a small section in the README or CHANGELOG explaining how to proceed with this change? This way you can just ask our partners to look at that section to understand how to proceed with the updated version of the event.

**Review by marcelosalloum — APPROVED** [review]
LGTM!

Plz take a look at [this comment](https://github.com/stellar/java-stellar-anchor-sdk/pull/698#discussion_r1062984236) before merging. Just a suggestion to make your life easier in the future 😄

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (lijamie98) [reply docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:14]: Oh yeah... definitely. I will add CHANGELOG which is closer to the change.

**Review by marcelosalloum — APPROVED** [review]
LGTM!

**Comment by lijamie98** [comment]
> I tried to design the `TransactionEvent` schema to be SEP-generic, can you describe what you think isn't generalized?
> 
> The only fields I see that could be confusing are `customers` & `creator`.
> 
> We needed `customers` for SEP-31, but in SEP-24 & 6 the customers (`customers.sender` and `customers.receiver`) were intended to be identical.
> 
> We needed `creator` for SEP-31 to save the send anchor's public key, but `creator` can also be used for the Stellar wallet. If wallet attribution is used we can put the wallet's home domain in the `creator` field.

These fields are available in the `transaction` payload.

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (JakeUrban) [inline docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:3]: How should this version relate to the AP's version? Should it match? Or should it be independent?

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (JakeUrban) [inline docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml:48]: Ah, very cool, I didn't know we could do this.

**Review by JakeUrban — COMMENTED** [review]
Hey Jamie, thanks for working on this. I think we should make a minor adjustment before merging.

Each event schema could have the same top level keys:
- `id` (this is the `event_id`)
- `type`
- `data`

The structure of `data` is dependent on `type`, and we can put the current `QuoteEvent` & `TransactionEvent` properties inside it. So for example:

For a transaction event:

```js
{
  "id": "1", // or whatever format we use for IDs
  "type": "transaction_created", // or something else transaction-related
  "data": {
    "transaction": {
      // all the other TransactionEvent properties
    }
  }
}
```

For quote events:

```js
{
  "id": "1",
  "type": "quote_created",
  "data": {
    "quote": {
      // all QuoteEvent properties, including the newly added 'transaction' object
    }
  }
}
```

File: docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml
Comment (lijamie98) [reply docs/03 - Implementing the Anchor Server/Communication/Events Schema.yml]: I would prefer independent. But all APIs should have the same version string.

However, I don't see the way to share the version string in swagger.

File: docs/03 - Implementing the Anchor Server/Communication/schemas.yml
Comment (JakeUrban) [inline docs/03 - Implementing the Anchor Server/Communication/schemas.yml:467]: Just double checking that you don't want to add a generic container key for the event payload? Something like `data` or `payload`?

Comment (lijamie98) [reply docs/03 - Implementing the Anchor Server/Communication/schemas.yml:467]: I think the generic container key `data` is not necessary. Do you have any good use case for the `data` field?

Comment (JakeUrban) [reply docs/03 - Implementing the Anchor Server/Communication/schemas.yml:467]: I've just seen it used in other event systems, like Circle's. Its not strictly necessary but it does provide a separation between the event scheme (`id`, `type`, & `data`) vs. the schema specific to the event type, which is inside `data`. If we ever had an event with lots of top-level attributes, then it would be nicer to have a container attribute.

I don't feel strongly though, so I'll leave it to you.

Comment (lijamie98) [reply docs/03 - Implementing the Anchor Server/Communication/schemas.yml:467]: Thanks Jake. Appreciated your insight very much. I guess we may always nest extra top-level attributes. My gut feeling is to keep it simple at this moment. 

We still have some time until we release the AP and we may always revisit this issue.

**Pull request merged**
